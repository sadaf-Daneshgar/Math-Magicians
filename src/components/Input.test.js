import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';

// Import the component you want to test
import InputNumber from './Input';

test('InputNumber component snapshot', () => {
  const props = {
    className: 'test-class',
    value: '123',
  };

  const tree = renderer.create(<InputNumber {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('InputNumber prop types validation', () => {
  const props = {
    className: 'test-class',
    value: '123',
  };

  const propTypesError = PropTypes.checkPropTypes(
    InputNumber.propTypes,
    props,
    'prop',
    InputNumber.name,
  );

  expect(propTypesError).toBeUndefined();
});
