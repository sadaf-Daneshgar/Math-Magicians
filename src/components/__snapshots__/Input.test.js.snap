// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNumber component snapshot 1`] = `
<div
  className="test-class"
>
  123
</div>
`;
